rocAL/rocAL/source/pipeline/master_graph.cpp

```diff
@@ -296,36 +296,37 @@ MasterGraph::build()
     start_processing();
     return Status::OK;
 }
-Image *
-MasterGraph::create_loader_output_image(const ImageInfo &info)
-{
+
+rocalTensor *
+MasterGraph::create_loader_output_tensor(const rocalTensorInfo &info) {
     /*
-    *   NOTE: Output image for a source node needs to be created as a regular (non-virtual) image
+    *   NOTE: Output tensor for a source node needs to be created as a regular (non-virtual) tensor
     */
-    auto output = new Image(info);
-
+    auto output = new rocalTensor(info);
     if(output->create_from_handle(_context) != 0)
-        THROW("Creating output image for loader failed");
+        THROW("Creating output tensor for loader failed");
 
-    _internal_images.push_back(output);
+    _internal_tensors.push_back(output);
 
     return output;
 }
 
-Image *
-MasterGraph::create_image(const ImageInfo &info, bool is_output)
-{
-    auto* output = new Image(info);
-    // if the image is not an output image, the image creation is deferred and later it'll be created as a virtual image
-    if(is_output)
-    {
-        if (output->create_from_handle(_context) != 0)
-            THROW("Cannot create the image from handle")
+rocalTensor *
+MasterGraph::create_tensor(const rocalTensorInfo &info, bool is_output) {
+    auto *new_tensor = new rocalTensor(info);
+    // if the tensor is not an output tensor, the tensor creation is deferred and later it'll be created as a virtual tensor
+    if(is_output) {
+        if (new_tensor->create_from_handle(_context) != 0)
```

Comment:
why do we need new_tensor and output?

Reply:
In image pipeline the output is returned to used by getting read buffers from ring buffer and copying the data into user passed buffers. In the current changes we are maintaining 2 tensor lists. The internal_tensor_list which is used to store a list of the output tensors and used in output routine(handle swapped with the write buffers in ring buffer) . The output_tensor_list which stores the same list of output tensors and is used to return the output tensor list to the user (handle swapped with the read buffers in ring buffer when output list is requested by the user).  Hence we are creating the new_tensor and output